pagopa/io-app
# chore(IT Wallet): [SIW-1869] Add fiscal code full screen modal

## Short description
This PR adds a feature that allows users to view their fiscal code in a full-screen modal. Users can access it by tapping the fiscal code on the TS credential details screen.

## List of changes proposed in this pull request
- Updated `react-native-screen-brightness` package patch removing unnecessary changes
- Updated locales
- Added `given_name` and `family_name` ids to the `WellKnowClaim` enumerator
- Added `extractClaim` utility function, which takes a claim id and an optional decoder, and returns a function which takes a `ParsedCredential` and tries to extract a claim value from it. Very useful when used in `fp-ts` contexts.
- Added `getFirstNameFromCredential` and `getFamilyNameFromCredential` functions
- Added `selectNameSurnameFromEid` selector, which extracts (if available) name and surname from the stored eID
- Updated `ItwPresentationFiscalCode` look and feel with a `Pressable`, including all the style and animations of a `Button`
- Added `ItwPresentationCredentialFiscalCodeModal` screen
- Fixed an issue with `useMaxBrightness` hook on Android, which was causing unwanted flashes during a brightness transition

## How to test
To test this PR you need to have an active and valid Wallet instance on your device and aTS credential.
1. Navigate to the TS details screen and verify that the Fiscal Code button is correctly displayed. 
2. Tap on the Fiscal code button and verify that the fiscal code is correctly shown in a full screen modal. 
3. Verify that the screen brightness is smoothly increased to the maximum when entering the full screen modal and then reverted back to the initial value when exiting. On Android, try different starting brightness levels and check that the transition is still working.

## Preview

| iOS | Android |
| --- | --- |
| <video src="https://github.com/user-attachments/assets/f1a0bb69-07a4-4b3e-8749-8651978265ef" /> | <video src="https://github.com/user-attachments/assets/2548e485-6a21-47db-9723-1613268c9304" /> |
